yarn-project/simulator/src/private/unconstrained_execution_oracle.ts

```diff
@@ -376,4 +377,23 @@ export class UnconstrainedExecutionOracle extends TypedOracle {
   public override getSharedSecret(address: AztecAddress, ephPk: Point): Promise<Point> {
     return this.executionDataProvider.getSharedSecret(address, ephPk);
   }
+
+  public override async storePrivateEventLog(
+    contractAddress: AztecAddress,
+    recipient: AztecAddress,
+    eventSelector: EventSelector,
+    logContent: Fr[],
+    txHash: TxHash,
+    logIndexInTx: number,
+  ): Promise<void> {
+    return this.executionDataProvider.storePrivateEventLog(
+      contractAddress,
+      recipient,
+      eventSelector,
+      logContent,
+      txHash,
+      logIndexInTx,
+      await this.executionDataProvider.getBlockNumber(),
```

Comment:
This is wrong, it should not be fetched from the exec data provider. I'd remove this param here and just get the correct block number (which is the one from the sync data provider) inside of pxe_oracle_interface directly. The fact that writing tests for this is very annoying/slow is a problem too.

Reply:
https://github.com/AztecProtocol/aztec-packages/pull/12957/commits/195c10d394fc593a651787b1d9e1f184b183ebb7